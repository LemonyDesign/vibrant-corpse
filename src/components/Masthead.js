import React from 'react';
import '../styles/base/base.scss';
import '../styles/components/masthead.scss';

function Masthead() {
  return (
    <header className="masthead">
      <h1 className="masthead__title">Vibrant Corpse</h1>
      <div className="masthead__description">
        <p>
          Vibrant corpse (
          <em>cadavre exquis</em>
          {' '}
in French) is a game invented by the Surrealists
          aimed at stimulating creative response. The method involves the assemblage of generated
          words or images, according to an agreed rule. This game seeks to mimic the original by
          randomly generating word lists then combining the results to create intriguing
          combinations that may be chosen for &lsquo;workshopping&rsquo; â€“ 5 minutes of free-flow
          writing.
        </p>
      </div>
    </header>
  );
}

export default Masthead;
