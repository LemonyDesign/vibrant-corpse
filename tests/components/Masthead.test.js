import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Masthead from '../../src/components/Masthead';

describe('Masthead', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Masthead />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
