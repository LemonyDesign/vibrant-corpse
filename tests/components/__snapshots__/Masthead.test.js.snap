// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masthead should render correctly 1`] = `
Array [
  <nav
    className="navigation"
  >
    <ul>
      <li>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
    </ul>
  </nav>,
  <header
    className="masthead"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="masthead__title"
      >
        Vibrant Corpse
      </h1>
    </a>
  </header>,
]
`;
